### PR TITLE
Add Gitea egg to repository

### LINKED ISSUE #1208: Gitea - Testing
Service: Gitea

Does this expand an already existing service: N

Link to game: https://gitea.io/en-us/

Links for server downloads: https://dl.gitea.io/gitea

ATTENTION: This is not a request, its an issue to test my created egg before it's pushed into the main repo

Test Egg: [egg-gi-tea.zip](https://github.com/parkervcp/eggs/files/6724261/egg-gi-tea.zip)

### PR DESCRIPTION
**Pull request by DatMayo**
### All Submissions:

* [X] Have you followed the guidelines in our Contributing document? `Yes`
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change? `Yes`
* [X] Did you branch your changes and PR from that branch and not from your master branch? `Yes`

### New Server Submissions:

1. [X] Does your submission pass tests (server is connectable)? `Yes`
2. [X] Does your server use a custom docker image? `No`
3. [X] Have you added the server to the main README.md? `Yes`
4. [X] Have you added a unique README.md for the server you are adding? `Yes`

Closes #1208

### CODE REVIEW

**Review by tmunsch — APPROVED** [review]
Works fine for me

File: README.md [hunk 1]
```diff
@@ -238,5 +238,7 @@ If you are reading this it looks like you are looking to add an egg to your serv
 * [Code-Server](/software/code-server)
 ### Grafana
 * [Grafana](/software/grafana)
+### Gitea
+* [Gitea](/software/gitea)
```
Comment (Software-Noob) [inline README.md:242]: Alphabetical order would put this above Grafana

Comment (DatMayo) [reply README.md:242]: Whoops, sry. I'll correct that within the next 24h

Comment (docimin) [reply README.md:242]: Any update @DatMayo?

File: software/gitea/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+[Gitea](https://gitea.io/en-us/) is a community managed lightweight code hosting solution written in Go. It is published under the MIT license.
+
+### Server Ports
+
+Ports required to run the server in a table format.
+
+| Port | default |
+| ---- | ------- |
+| App  | 3000    |
+| SSH  | 22      |
```
Comment (Software-Noob) [inline software/gitea/README.md:11]: This is below the possible Pterodactyl port range - 1024 is the minimum.

File: software/gitea/egg-gitea.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            "rules": "required|string|in:true,false"
+        },
+        {
+            "name": "SSH Port",
+            "description": "SSH port displayed in clone URL.",
+            "env_variable": "SSH_PORT",
+            "default_value": "22",
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "required|integer|between:1,65535"
```
Comment (Software-Noob) [inline software/gitea/egg-gitea.json:47]: This is below the possible Pterodactyl port range - 1024 is the minimum.

**Review by Software-Noob — CHANGES_REQUESTED** [review]

**Review by Software-Noob — APPROVED** [review]
I've fixed the conflicts myself and tested using reverse proxy. Thanks for the contribution, LGTM.

**Comment by DatMayo** [comment]
@Software-Noob thx for your help. I didn't had the time to fix

**Pull request merged**